Frameworks/CoreGraphics/D2DWrapper.mm

```diff
@@ -0,0 +1,33 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <CoreGraphics/D2DWrapper.h>
+#import <Starboard.h>
+
+static const wchar_t* TAG = L"_D2DWrapper";
+
+// Helper for creating a D2DFactory
+static ComPtr<ID2D1Factory> _CreateD2DFactoryInstance(D2D1_FACTORY_TYPE type) {
+    ComPtr<ID2D1Factory> d2dFactory;
+    THROW_IF_FAILED(D2D1CreateFactory(type, d2dFactory.ReleaseAndGetAddressOf()));
```

Comment:
D2D1CreateFactory doesn't work well with a ComPtr since it's not typed correctly. There are two ways to work with this then, tell D2D1 the actual type of what's being passed in with __uuidof(ID2D1Factory) or to pass something with the proper type. It might be better to just use the uuid here though for clarity.

---
In reply to: [82041278](https://github.com/Microsoft/WinObjC/pull/1085#discussion_r82041278) [](ancestors = 82041278)